src/hooks/transactions/useCheckTransactionStatus/useCheckTransactionStatus.ts

```diff
@@ -0,0 +1,197 @@
+import { useRef } from 'react';
+import { getTransactionsByHashes as defaultGetTxByHash } from 'apiCalls/transactions';
+import { GetTransactionsByHashesType } from 'contexts/types';
+import { useDispatch } from 'redux/DappProviderContext';
+import {
+  updateSignedTransactions,
+  updateSignedTransactionStatus
+} from 'redux/slices';
+import { useGetPendingTransactions } from 'services';
+import {
+  TransactionBatchStatusesEnum,
+  TransactionServerStatusesEnum
+} from 'types/enums';
+import {
+  SignedTransactionsBodyType,
+  SmartContractResult
+} from 'types/transactions';
+import { getIsTransactionFailed, getIsTransactionPending } from 'utils';
+import { refreshAccount } from 'utils/account';
+import getPendingTransactions from './getPendingTransactions';
+
+interface TransactionStatusTrackerPropsType {
+  sessionId: string;
+  transactionBatch: SignedTransactionsBodyType;
+  getTransactionsByHash?: GetTransactionsByHashesType;
+  refetchTimestamp?: number;
+}
+
+interface RetriesType {
+  [hash: string]: number;
+}
+
+function useManageFailed() {
+  const dispatch = useDispatch();
```

Comment:
This seems to be a hook only because of useDispatch, it will probably be cleaner to just use store.dispatch and make this a regular function

Reply:
✅